--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -23,6 +23,14 @@
 
 #ifdef CONFIG_IOCTL_CFG80211
 
+#include <rtw_wifi_regd.h>
+
+#ifdef BACKPORT
+#pragma push_macro("LINUX_VERSION_CODE")
+#undef LINUX_VERSION_CODE
+#define LINUX_VERSION_CODE KERNEL_VERSION(4,6,0)
+#endif
+
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 0, 0))
 #define STATION_INFO_SIGNAL		BIT(NL80211_STA_INFO_SIGNAL)
 #define STATION_INFO_TX_BITRATE		BIT(NL80211_STA_INFO_TX_BITRATE)
@@ -31,8 +39,6 @@
 #define STATION_INFO_ASSOC_REQ_IES	0
 #endif /* Linux kernel >= 4.0.0 */
 
-#include <rtw_wifi_regd.h>
-
 #define RTW_MAX_MGMT_TX_CNT (8)
 #define RTW_MAX_MGMT_TX_MS_GAS (500)
 
@@ -6951,5 +6957,8 @@ void rtw_cfg80211_dev_res_unregister(str
 #endif
 }
 
-#endif /* CONFIG_IOCTL_CFG80211 */
+#ifdef BACKPORT
+#pragma pop_macro("LINUX_VERSION_CODE")
+#endif
 
+#endif /* CONFIG_IOCTL_CFG80211 */
--- a/os_dep/linux/rtw_cfgvendor.c
+++ b/os_dep/linux/rtw_cfgvendor.c
@@ -42,6 +42,12 @@
 
 #include <net/rtnetlink.h>
 
+#ifdef BACKPORT
+#pragma push_macro("LINUX_VERSION_CODE")
+#undef LINUX_VERSION_CODE
+#define LINUX_VERSION_CODE KERNEL_VERSION(4,6,0)
+#endif
+
 #ifdef DBG_MEM_ALLOC
 extern bool match_mstat_sniff_rules(const enum mstat_f flags, const size_t size);
 struct sk_buff *dbg_rtw_cfg80211_vendor_event_alloc(struct wiphy *wiphy, int len, int event_id, gfp_t gfp
@@ -1336,5 +1342,9 @@ int rtw_cfgvendor_detach(struct wiphy *w
 }
 #endif /* (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)) || defined(RTW_VENDOR_EXT_SUPPORT) */
 
+#ifdef BACKPORT
+#pragma pop_macro("LINUX_VERSION_CODE")
+#endif
+
 #endif /* CONFIG_IOCTL_CFG80211 */
 
