--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -23,8 +23,15 @@
 
 #ifdef CONFIG_IOCTL_CFG80211
 
+#ifdef BACKPORT
+#pragma push_macro("LINUX_VERSION_CODE")
+#undef LINUX_VERSION_CODE
+#define LINUX_VERSION_CODE KERNEL_VERSION(4,8,0)
+#endif
+
 #include <rtw_wifi_regd.h>
 
+
 #define RTW_MAX_MGMT_TX_CNT (8)
 
 #define RTW_SCAN_IE_LEN_MAX      2304
@@ -6435,5 +6442,8 @@ void rtw_wdev_unregister(struct wireless
 	wiphy_unregister(wdev->wiphy);
 }
 
-#endif //CONFIG_IOCTL_CFG80211
+#ifdef BACKPORT
+#pragma pop_macro("LINUX_VERSION_CODE")
+#endif
 
+#endif /* CONFIG_IOCTL_CFG80211 */
--- a/os_dep/linux/rtw_cfgvendor.c
+++ b/os_dep/linux/rtw_cfgvendor.c
@@ -42,6 +42,12 @@
 
 #include <net/rtnetlink.h>
 
+#ifdef BACKPORT
+#pragma push_macro("LINUX_VERSION_CODE")
+#undef LINUX_VERSION_CODE
+#define LINUX_VERSION_CODE KERNEL_VERSION(4,8,0)
+#endif
+
 #ifdef DBG_MEM_ALLOC
 extern bool match_mstat_sniff_rules(const enum mstat_f flags, const size_t size);
 struct sk_buff * dbg_rtw_cfg80211_vendor_event_alloc(struct wiphy *wiphy, int len, int event_id, gfp_t gfp
@@ -1335,5 +1341,9 @@ int rtw_cfgvendor_detach(struct wiphy *w
 }
 #endif /* (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)) || defined(RTW_VENDOR_EXT_SUPPORT) */
 
+#ifdef BACKPORT
+#pragma pop_macro("LINUX_VERSION_CODE")
+#endif
+
 #endif /* CONFIG_IOCTL_CFG80211 */
 
--- a/include/rtw_wifi_regd.h
+++ b/include/rtw_wifi_regd.h
@@ -22,12 +22,10 @@ enum country_code_type_t {
 int rtw_regd_init(_adapter *padapter);
 void rtw_reg_notify_by_driver(_adapter *adapter);
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 7, 0))
 #define ieee80211_band nl80211_band
 #define IEEE80211_BAND_2GHZ NL80211_BAND_2GHZ
 #define IEEE80211_BAND_5GHZ NL80211_BAND_5GHZ
 #define IEEE80211_NUM_BANDS NUM_NL80211_BANDS
-#endif
 
 #endif /* __RTW_WIFI_REGD_H__ */
 
