--- a/hal/phydm/phydm_dig.c
+++ b/hal/phydm/phydm_dig.c
@@ -567,7 +567,7 @@ odm_PauseDIG(
 		}
 
 		if (BIT(pause_level) > pDM_DigTable->pause_dig_level) {
-			u1Byte		max_level;
+			s1Byte		max_level;
 		
 			/* Calculate the maximum level now */
 			for (max_level = (pause_level - 1); max_level >= 0; max_level--) {
@@ -1731,7 +1731,7 @@ odm_PauseCCKPacketDetection(
 		}
 
 		if (BIT(pause_level) > pDM_DigTable->pause_cckpd_level) {
-			u1Byte	max_level;
+			s1Byte	max_level;
 		
 			/* Calculate the maximum level now */
 			for (max_level = (pause_level - 1); max_level >= 0; max_level--) {
--- a/hal/phydm/phydm_beamforming.c
+++ b/hal/phydm/phydm_beamforming.c
@@ -109,7 +109,7 @@ Beamforming_GetHTNDPTxRate(
 )
 {
 	PDM_ODM_T	pDM_Odm = (PDM_ODM_T)pDM_VOID;
-	u1Byte Nr_index = 0;
+	s1Byte Nr_index = 0;
 	u1Byte NDPTxRate;
 	/*Find Nr*/
 	
@@ -147,7 +147,7 @@ Beamforming_GetVHTNDPTxRate(
 )
 {
 	PDM_ODM_T	pDM_Odm = (PDM_ODM_T)pDM_VOID;
-	u1Byte Nr_index = 0;
+	s1Byte Nr_index = 0;
 	u1Byte NDPTxRate;
 	/*Find Nr*/
 	if (pDM_Odm->SupportICType & ODM_RTL8814A)
--- a/hal/phydm/txbf/halcomtxbf.h
+++ b/hal/phydm/txbf/halcomtxbf.h
@@ -16,7 +16,11 @@ typedef	BOOLEAN
 	OUT	PVOID			pInBuf
 	);
 */
+#ifdef CONFIG_RTL8814A
 #define TxBF_Nr(a, b) ((a > b) ? (b) : (a))
+#else
+#define TxBF_Nr(a, b) (a)
+#endif
 
 typedef enum _TXBF_SET_TYPE{
 	TXBF_SET_SOUNDING_ENTER,
--- a/core/rtw_ap.c
+++ b/core/rtw_ap.c
@@ -202,7 +202,7 @@ void rtw_add_bcn_ie(_adapter *padapter,
 	u8	bmatch = _FALSE;
 	u8	*pie = pnetwork->IEs;
 	u8	*p=NULL, *dst_ie=NULL, *premainder_ie=NULL, *pbackup_remainder_ie=NULL;
-	u32	i, offset, ielen, ie_offset, remainder_ielen = 0;
+	u32	i, offset, ielen = 0, ie_offset, remainder_ielen = 0;
 
 	for (i = sizeof(NDIS_802_11_FIXED_IEs); i < pnetwork->IELength;)
 	{
