--- a/core/rtw_sta_mgt.c
+++ b/core/rtw_sta_mgt.c
@@ -336,10 +336,14 @@ _func_enter_;
 
 	//_enter_critical_bh(&(pfree_sta_queue->lock), &irqL);
 	_enter_critical_bh(&(pstapriv->sta_hash_lock), &irqL2);
-	if (_rtw_queue_empty(pfree_sta_queue) == _TRUE)
+	if (!pstapriv->padapter->pnetdev)
+	{
+		psta = NULL;
+	}
+	else if (_rtw_queue_empty(pfree_sta_queue) == _TRUE)
 	{
 		//_exit_critical_bh(&(pfree_sta_queue->lock), &irqL);
-		_exit_critical_bh(&(pstapriv->sta_hash_lock), &irqL2);
+		//_exit_critical_bh(&(pstapriv->sta_hash_lock), &irqL2);
 		psta = NULL;
 	}
 	else
